src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/NamespaceAction.java

```diff
@@ -10,6 +10,10 @@
 import hudson.model.Run;
 import jenkins.model.RunAction2;
 
+/**
+ * Use <pre>getContext().get(PodTemplateContext.class)</pre> instead.
+ */
+@Deprecated
 public class NamespaceAction extends AbstractInvisibleRunAction2 implements RunAction2 {
```

Comment:
`AbstractInvisibleRunAction2` should also be deprecated, as you are deprecating both of its subtypes.